python_bindings/bindings.cpp

```diff
@@ -74,60 +74,97 @@ inline void ParallelFor(size_t start, size_t end, size_t numThreads, Function fn
 
 
 inline void assert_true(bool expr, const std::string & msg) {
-    if (expr == false)
-    throw std::runtime_error("Unpickle Error: "+msg);
+    if (expr == false) throw std::runtime_error("Unpickle Error: " + msg);
     return;
 }
 
 
-template<typename dist_t, typename data_t=float>
-class Index {
-public:
-  Index(const std::string &space_name, const int dim) :
-  space_name(space_name), dim(dim) {
-    normalize=false;
-    if(space_name=="l2") {
-      l2space = new hnswlib::L2Space(dim);
-    }
-    else if(space_name=="ip") {
-      l2space = new hnswlib::InnerProductSpace(dim);
-    }
-    else if(space_name=="cosine") {
-      l2space = new hnswlib::InnerProductSpace(dim);
-      normalize=true;
+inline void set_input_array_shapes(const py::buffer_info& buffer, size_t* rows, size_t* features) {
```

Comment:
Sound like it should be get_array_shape or something.

Reply:
Will be addressed in next PR